### PR TITLE
Front controller implementation

### PR DESCRIPTION
**Pull request by jeremyfifty9**
I've created a front controlling mechanism to make the bootstrap loading more object-oriented.

Looking for feedback on the file structure & how I tie it all together/the architecture.

### CODE REVIEW

**Comment by jeremyfifty9** [comment]
Issue that I just noticed is that `bootstrap-http.php` is used by the API. so before pulling I ned to find a way to remove that dependency

**Comment by jeremyfifty9** [comment]
@Terf those lines were not directly deleted, but removed because the actual instantiation of the controller and view was outsourced to the Dispatcher. in the bootstrap now is:

https://github.com/EnvironmentalDashboard/community-voices/blob/3f94c8490c31bfc9bf85997c9e2adab91d73aed2/src/CommunityVoices/App/Website/bootstrap.php#L31-L35

I think the comments you were talking about were attached to:

https://github.com/EnvironmentalDashboard/community-voices/blob/3f94c8490c31bfc9bf85997c9e2adab91d73aed2/src/CommunityVoices/App/Website/Bootstrap/Dispatcher.php#L26-L30

I can add more documentation to the bootstrap if you think it's necessary.

I'll change `$appPrefix`

File: src/CommunityVoices/App/Website/Bootstrap/FrontController.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+class FrontController {
+    protected $router;
+    protected $dispatcher;
+    protected $injector;
+
+    /**
+     * Providers to initialize prior to request route & dispatching
+     * 
+     * Note that the routes provider is excluded because it sould be initialized
```
Comment (Sammidysam) [inline src/CommunityVoices/App/Website/Bootstrap/FrontController.php]: *should

File: src/CommunityVoices/App/Website/Bootstrap/Router.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+         * @config
+         * @todo
+         */
+        $production_server = (getenv('SERVER') === 'environmentaldashboard.org');
+        
+        $uri = isset($_SERVER['REQUEST_URI'])
+                    ? $_SERVER['REQUEST_URI']
+                    : '/';
+
+        $uri = ($production_server) ? '/community-voices' . substr(explode('?', $uri)[0], 1) : explode('?', $uri)[0];
```
Comment (Sammidysam) [inline src/CommunityVoices/App/Website/Bootstrap/Router.php]: Could we add more parentheses to make this more clearly obvious?  I don't ever want to have to think about order of operations.

**Comment by Sammidysam** [comment]
What ideas did you have for changing `$appPrefix`?  I feel like we kind of need it stored in the application too because otherwise how are we going to be able to build the application such that for most intents and purposes the programmer never needs to know that the prefix is present?  I figure if we build our deepest layer of infrastructure with handling the prefix, it could make every upper layer much more simple to manage.

**Comment by Sammidysam** [comment]
Overall, this looks great though and I'm glad you put this work in. :+1:

File: src/CommunityVoices/App/Website/Bootstrap/Router.php (hunk 2 above)
Comment (jeremyfifty9) [reply src/CommunityVoices/App/Website/Bootstrap/Router.php]: @Terf is the ternary condition still necessary here if we remove the appPrefix and change the etc conf

File: src/CommunityVoices/App/Website/Bootstrap/FrontController.php (hunk 1 above)
Comment (jeremyfifty9) [reply src/CommunityVoices/App/Website/Bootstrap/FrontController.php]: thanks. I need tor remove that anyways since routes isn't a provider anymore

**Comment by jeremyfifty9** [comment]
what do you mean by https://github.com/EnvironmentalDashboard/community-voices/pull/22#issuecomment-464286757? I'm guessing app prefix will be part of the central configuration you build.

**Comment by Sammidysam** [comment]
Well in https://github.com/EnvironmentalDashboard/community-voices/pull/22#issuecomment-464102913 you said "I'll change `$appPrefix`"

**Comment by jeremyfifty9** [comment]
@Sammidysam i wasn't clear enough / tim and I talked about it later, so there's more to it: appPrefix will still be there, but by default it will be empty (and eventually worked into a dotenv)

**Comment by jeremyfifty9** [comment]
to do

- app prefix
- api routing
- figure out url generator providing

**Pull request merged**